File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
import java.util.*;

class Solution {
    public int solution(String t, String p) {
        
        int t_size = t.length(); // t의 길이
        int p_size = p.length(); // p의 길이
        long p_value = Long.parseLong(p); // 비교할 숫자
            
        List<Long> part_number = new ArrayList<>();
        
        for(int i = 0; i < t_size - p_size + 1; i++){
            part_number.add(Long.parseLong(t.substring(i,i + p_size)));
        }
        
        int answer = 0;
        
        for(long num : part_number){
            if(num <= p_value){
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    
    size = len(p) # p의 자리수
    
    part_numbers = []
    
    for i in range(len(t) - size + 1):
        part_numbers.append(t[i:i+size])
    
    answer = 0
    
    for number in part_numbers:
        if int(number) <= int(p):
            answer += 1
    return answer